Hibernate/hibernateAnnotationDemo/PrimaryKeyDemo.java

```diff
@@ -0,0 +1,32 @@
+package hibernateAnnotationDemo;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.cfg.Configuration;
+
+public class PrimaryKeyDemo {
+	public static void main(String[] args) {
+		SessionFactory factory = new Configuration()
+				.configure("hibernate.cfg.xml")
+				.addAnnotatedClass(Student.class)
+				.buildSessionFactory();
+
+		Session session = factory.getCurrentSession();
+		try {			
+			System.out.println("Creating new student object...");
+			Student tempStudent = new Student("Ram", "kumar", "ram@gmail.com");
+			Student tempStudent1 = new Student("John", "Doe", "John@gmail.com");
```

Comment:
try inserting tempStudent two times and see what happens ?

Reply:
same data will be inserted twice with different id's.May be by using saveOrUpdate instead of savewe can rectify duplicate entries